apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php

```diff
@@ -147,4 +114,26 @@ public function enqueue_styles() {
 			filemtime( plugin_dir_path( __FILE__ ) . 'dist/' . $style_file )
 		);
 	}
+
+	/**
+	 * Fetch vertical data from the API or return cached version if available.
+	 */
+	public function fetch_vertical_data() {
+		$vertical_templates = get_transient( self::$transient_key );
+		$vertical_id        = get_site_option( 'site_vertical', 'default' );
+
+		// Load fresh data if we don't have any or vertical_id doesn't match.
+		if ( false === $vertical_templates || $vertical_id !== $vertical_templates['vertical_id'] ) {
+			$request_url = 'https://public-api.wordpress.com/wpcom/v2/verticals/' . $vertical_id . '/templates';
+			$response    = wp_remote_get( esc_url_raw( $request_url ) );
+			if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+				return false;
```

Comment:
What do you think about returning an empty array here to keep the return type consistent?